Package

rickhelmus opened the pull request:
Hello,

This PR makes a few small changes to convert c3sdb into a package structure, so it can easily be installed with pip. The  pretrained/ directory was moved inside c3sdb/ so the pickled files will be added into the package. A small utility function (`pretrained_data()`) was added to access these files once the package is installed.

Some of the package metadata could probably be filled in better. Furthermore, I am not very experienced with Python, so feel free to point out mistakes ;-)

rickhelmus committed:
initial commit for package structure

* added basic pyproject.toml and setup.cfg
* moved pretrained folder inside c3sdb/ so it will be packaged
* added small utitlity function to access the pretrained directory: pretrained_data()

rickhelmus committed:
remove unneeded import

dylanhross committed:
Update setup.cfg

Take the version directly from the package itself

dylanhross commented:
I edited some of the metadata items in `setup.cfg`. My only remaining concern is whether you have tested that the code still works with this installable package structure? I am specifically thinking about [this section](https://github.com/dylanhross/c3sdb?tab=readme-ov-file#inference-with-trained-model) which is the example for how to perform model inference using the pre-trained parameters. With the pre-trained model parameter files now bundled into the package and accessible via a function, I suggest that you : 
1. test that inference still works, following that example from the README
2. update that section of the README with a new code example that uses the `pretrained_data()` function

rickhelmus committed:
update installation and inference example

rickhelmus commented:
Hi Dylan,

Thanks for the fast feedback! I did test (once more) the examples, and updated the `README.md` accordingly. I also modified the installation instructions. Please let me know if you feel anything else needs to be addressed.

Thanks,
Rick
